client/src/components/Cart/Cart.jsx

```diff
@@ -1,29 +1,89 @@
+/* eslint-disable react/prop-types */
 import React, { useEffect, useState } from 'react';
 import Swal from 'sweetalert2';
 import Empty from './Empty';
 
 import {
-  CartItem, CartContainer, CartItemPrice, CartItemTitle, CartTitle, RemoveButton,
+  CartItem,
+  CartContainer,
+  CartItemPrice,
+  CartItemTitle,
+  CartTitle,
+  RemoveButton,
+  Image,
+  CartItemCount,
+  Button,
+  ClearButton,
+  GlobalStyle,
 } from './Style.css';
 
-function Cart() {
+function Cart({ userId, productId }) {
   const [cartItems, setCartItems] = useState([]);
 
   useEffect(() => {
-    fetch('/cart')
+    fetch(`/cart/${userId}`)
       .then((response) => response.json())
       .then((data) => setCartItems(data))
       .catch((error) => {
-        console.error(error);
         Swal.fire({
           icon: 'error',
           title: 'An error occurred',
           text: error.message,
         });
       });
   }, []);
+  const clearCart = () => {
+    Swal.fire({
+      icon: 'question',
+      title: 'Confirmation',
+      text: 'Are you sure you want to clear your cart?',
+      showCancelButton: true,
+      confirmButtonText: 'Remove',
+      cancelButtonText: 'Cancel',
+      reverseButtons: true,
+    }).then((result) => {
+      if (result.isConfirmed) {
+        fetch(`/clearCart/${userId}`, {
```

Comment:
Here you shouldn't make a promise hell, `then` methods must be after each other not nested to each other.